omod/src/main/java/org/openmrs/module/smartonfhir/web/filter/SmartSessionLogoutFilter.java

```diff
@@ -34,7 +35,15 @@
 	
 	@Override
 	public void init(FilterConfig filterConfig) {
-		logoutUrl = KeycloakDeploymentBuilder.build(KeycloakConfigHolder.getKeycloakConfig()).getLogoutUrl().toTemplate();
+		final KeycloakUriBuilder keycloakUriBuilder = KeycloakDeploymentBuilder
+				.build(KeycloakConfigHolder.getKeycloakConfig()).getLogoutUrl();
+
+		if (keycloakUriBuilder != null) {
+			logoutUrl = keycloakUriBuilder.toTemplate();
+		} else {
+			log.error(
+					"Could not find Keycloak configuration file. Please run keycloak server before openmrs to avoid this error");
```

Comment:
thanks @theanandankit  did this formatting get added automatically?

Reply:
@HerbertYiga Actually, to get the logoutURL for ending the keycloak session we need to import the keycloak config which later sends the GET request to the keycloak server to end the session. So we need to start the keycloak server first to get the keycloak config that contains the Logout URL.